streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -646,8 +645,8 @@ public void concurrentAccesses() throws Exception {
         producerThread.start();
 
         try {
-            waitUntilAtLeastNumRecordProcessed(outputTopicConcurrent, numberOfWordsPerIteration);
-            waitUntilAtLeastNumRecordProcessed(outputTopicConcurrentWindowed, numberOfWordsPerIteration);
+            waitUntilAtLeastNumRecordProcessed(outputTopicConcurrent, 1);
```

Comment:
This test becomes more flaky than before after the main loop refactoring, shown as the following:

```
java.lang.AssertionError: Did not receive all 48 records from topic output-concurrent-2 within 120000 ms
Expected: is a value equal to or greater than <48>
     but: <38> was less than <48>
```

Each iteration has 48 words, and 38 unique words, and when it fails it reported only 38 records ever consumed from the output topic. I suspect this is because of the caching effects, we processed all 500000 iterations without sending out any updates to the resulted count table output topic (seems we process too fast?). 

I piggy-backed a fix for this flakiness by reducing the requirement that we start the verification as long as we've seen one record in the output, similar as we do in other tests, and then in verifyGreaterOrEqual we do not require all keys been populated already.

Reply:
prop: I would prefer to have this fix in its own PR, because the commit would be easier to retrieve and to reference on failure tickets.